### DISCUSSION TITLE
Possible to have a kind of rowspan in a datagrid.

### DISCUSSION THREAD

**Discussion by well0549** [Megabit/Blazorise#4726] [General]
I have a datagrid with around 13 columns.
All the columns but 1 have a single short string in it.

But one column could have several lines of text in it. I hav etried several things to keep the string on one line.
I know this can be done with the style.

What i want is.....
1. To have the gird fill the whole screen (it's doing that)
2. To show as much as possible of the multiline text (on one line)

I have tried several things but the only thing that worked  a bit was setting the max width.

`                <DataGridColumn Field="@nameof(ClaimPointerAudit.EventText)" Caption="Event text">
                    <DisplayTemplate>
                        <Div Style="min-width:80px;max-width: 600px;white-space:nowrap;overflow:hidden;text-overflow: ellipsis">
                            @context.EventText
                        </Div>
                    </DisplayTemplate>
                </DataGridColumn>
`
But then it would never go beyond the 600 pix (and that's a waste in my 38inch monitor)

So i want to show as much of the text as possible and without giving tany of the columns a width.
all the columns should just be displayed but the column with the text in it should stretch as much as possible.

Maybe it's already possible but i hav not figured it ou yet. Any help would be apreciated.

**Comment by stsrki** [comment]
I don't think it would be possible to do it at the moment. There is no way to control the `td` outside of DataGrid. We would have to add new APIs on DataGridColumn

- `RowSpan`, and
- `ColumnSpan`

It is not such a bad idea to have.

Also, it seems we lack an `Attributes` parameter to catch all custom HTML attributes missing from our API. I will open a new ticket to add it to the next milestone.

**Reply by well0549** [reply]
Sounds promising

**Reply by David-Moreira** [reply]
However try something like
You could set it to Responsive, which should make it so, it tries to accomodate all text, but instead has an horizontal scrollbar, other than that I think you'd have to apply a general style to your grids so they would turn apply elipsis to the data.
So for example, if it's just some columns you can do something like:
`                                    <DisplayTemplate>
                                            <Paragraph Overflow="Overflow.Hidden" TextOverflow="TextOverflow.Truncate" CopyToClipboard>
                                                @context.HugeMessage
                                            </Paragraph>
                                    </DisplayTemplate>`

**Comment by steal4life** [comment]
After conducting thorough testing, we have concluded that adding parameters such as row span and column span to the Blazorise Datagrid does not make sense. In fact, it can even break the functionality of the Datagrid.

The Datagrid is designed to display tabular data in a structured and organized manner. Each cell in the Datagrid represents a unique piece of data, and modifying the row span or column span would disrupt this structure.

Furthermore, the Datagrid provides several powerful features for sorting, filtering, and manipulating data, all of which rely on the consistent and predictable structure of the Datagrid. Adding row span and column span would make it much more difficult to implement these features and could lead to unexpected behavior.

Therefore, based on our testing and analysis, we recommend against adding row span and column span to the Blazorise Datagrid. Instead, we suggest exploring other alternatives that are better suited to the task at hand.

![image](https://user-images.githubusercontent.com/79018897/232739301-80bab578-eec3-43be-94f9-8706f41e2a01.png)
